torch/_meta_registrations.py

```diff
@@ -2937,10 +2937,34 @@ def meta_addbmm(self, batch1, batch2, *, beta=1, alpha=1):
 @register_meta(
     [
         aten._foreach_abs_.default,
+        aten._foreach_acos_.default,
+        aten._foreach_asin_.default,
+        aten._foreach_ceil_.default,
+        aten._foreach_cos_.default,
+        aten._foreach_cosh_.default,
+        aten._foreach_erf_.default,
+        aten._foreach_erfc_.default,
+        aten._foreach_exp_.default,
+        aten._foreach_expm1_.default,
+        aten._foreach_frac_.default,
+        aten._foreach_floor_.default,
+        aten._foreach_lgamma_.default,
+        aten._foreach_log_.default,
+        aten._foreach_log10_.default,
+        aten._foreach_log1p_.default,
+        aten._foreach_log2_.default,
         aten._foreach_neg_.default,
         aten._foreach_reciprocal_.default,
-        aten._foreach_sqrt_.default,
+        aten._foreach_round_.default,
+        aten._foreach_sigmoid_.default,
         aten._foreach_sign_.default,
+        aten._foreach_sin_.default,
+        aten._foreach_sinh_.default,
+        aten._foreach_sqrt_.default,
+        aten._foreach_tan_.default,
+        aten._foreach_tanh_.default,
+        aten._foreach_trunc_.default,
+        aten._foreach_zero_.default,
     ]
 )
 def meta__foreach_unaop_(self):
```

Comment:
Preexisting, but this implementation looks wrong. A proper implementation would call the relevant pointwise op in a for loop for all the inputs. Same for all the other foreach ops.

Reply:
Updated to use the pointwise op.